### ISSUE TITLE
PLEASE don't use git flow

### ISSUE THREAD

**Issue by ferdnyc** [OpenShot/libopenshot#103]
I noticed from someone's logs that Openshot 2.4.2-rc1 was using "libopenshot 0.2.0", which is weird because my local builds are still calling themselves 0.1.9 and I'm fully updated with the master branch.

So then I went poking around and eventually found 2db0d91 "Updating gitlab CI to support git flow (develop, release, master + tag)". Sure enough, there's now a `libopenshot/release` branch which bumps the version number, while `libopenshot/master` is still at 0.1.9. (Weirdly, `libopenshot/develop` is also still at 0.1.9, which doesn't sound like correct use of git flow either.)

Git flow was originally proposed in 2010, and it was widely derided specifically because under its model master no longer contains the most recent code. Gitlab, in fact, even have a section titled "Git flow and its problems" in the documentation on their own [GitLab Flow](https://docs.gitlab.com/ee/workflow/gitlab_flow.html) model:

<blockquote>
Git flow was one of the first proposals to use git branches and it has gotten a lot of attention. It advocates a master branch and a separate develop branch as well as supporting branches for features, releases and hotfixes. The development happens on the develop branch, moves to a release branch and is finally merged into the master branch. Git flow is a well defined standard but its complexity introduces two problems. The first problem is that developers must use the develop branch and not master, master is reserved for code that is released to production. It is a convention to call your default branch master and to mostly branch from and merge to this. Since most tools automatically make the master branch the default one and display that one by default it is annoying to have to switch to another one. The second problem of git flow is the complexity introduced by the hotfix and release branches. These branches can be a good idea for some organizations but are overkill for the vast majority of them. Nowadays most organizations practice continuous delivery which means that your default branch can be deployed. This means that hotfix and release branches can be prevented including all the ceremony they introduce. An example of this ceremony is the merging back of release branches. Though specialized tools do exist to solve this, they require documentation and add complexity. Frequently developers make a mistake and for example changes are only merged into master and not into the develop branch. The root cause of these errors is that git flow is too complex for most of the use cases. And doing releases doesn't automatically mean also doing hotfixes.</blockquote>

**Comment by ferdnyc** [comment]
See also: [A succesful Git branching model considered harmful](https://barro.github.io/2016/02/a-succesful-git-branching-model-considered-harmful/).

**Comment by ferdnyc** [comment]
Meh, I'm over it. 

GitHub fortunately works very hard to paper over Git Flow's worst transgressions. And while it's still _very_ annoying when existing upstreams convert to Flow (all forks have to manually jump through hoops to get themselves reconfigured along the same lines), it's tolerable in repos that have been set up with develop-as-default.

**Comment by peanutbutterandcrackers** [comment]
@ferdnyc - v2.4.2 is out, BTW.

Also, do you have any other flow that you think will work best for OpenShot, good sir? (You always have cool ideas)

**Comment by ferdnyc** [comment]
> Also, do you have any other flow that you think will work best for OpenShot, good sir? (You always have cool ideas)

I started making the mistake of actually answering this question, but the truth is it doesn't matter. The time to explore alternatives or consider a different model would've been right at the beginning (or ideally even before that). Not weeks later, when significant effort has already been expended on the adoption of Git Flow. 

If there's any idea I've tried to champion, it's the importance of making choices and committing to them, not  switching lanes every time some new idea is brought up.
